spec/models/submission_file_spec.rb

```diff
@@ -0,0 +1,314 @@
+require 'spec_helper'
+
+describe SubmissionFile do
+  # Ensure that the the following relationship exists
+  context 'An existing SubmissionFile' do
+    it { is_expected.to belong_to(:submission) }
+    it { is_expected.to have_many(:annotations) }
+    it { is_expected.to validate_presence_of :submission }
+    it { is_expected.to validate_presence_of :filename }
+    it { is_expected.to validate_presence_of :path }
+  end
+
+  context 'A SubmissionFile' do
+
+    context 'A SubmissionFile without parameter and without id' do
```

Comment:
You can remove this block of tests (down to line 72); it's redundant with the 5-11 block.